### ISSUE TITLE
feat: bulk CIDs import

### ISSUE THREAD

**Issue by asigalov61** [ipfs/ipfs-webui#2173]
Hey guys,

I was wondering if it would be possible to add a bulk files/CIDs import from a txt or csv file?

I have a list of CIDs but there is no option in IPFS-Desktop to import them in bulk/automatically.

Can you implement this option? Or, if it is possible to do via Kubo or some other script method, it would be nice to see such method in the docs.

Please let me know.

Alex

**Comment by asigalov61** [comment]
@SgtPooki Yes, sorry about closing it. I would love to see the native support for bulk CID import in IPSF desktop :) Thank you, guys, for your hard work. IPFS is an important and needed thing :)

**Comment by asigalov61** [comment]
@SgtPooki Dear Sgt Pooki,

Any update on this feature? Any ETA on when it will be implemented?

Thank you so much in advance  for your response.

Alex

**Comment by SgtPooki** [comment]
@asigalov61 unfortunately, we've got a lot of items on our plate now, and a smaller team. I would be happy to review a PR that added a "bulk import" option to the Import button on the files screen: 

<img width="264" alt="image" src="https://github.com/ipfs/ipfs-webui/assets/1173416/b8e2d0bf-a7fc-49b8-b1d4-eabbdfe95cd1">

but won't be able to work on this for the forsee-able future.

**Comment by asigalov61** [comment]
@SgtPooki Thank you for your response and update :)

Unfortunately, IPFS code is too complex for me to contribute to. Sorry. I am mostly Python ML/DL dev :(

But please add this feature when you are able. I think it would be a useful addition to IPFS.

I will use your workaround for now, for which I am grateful nevertherless :)

Sincerely,

Alex.

**Comment by itsib** [comment]
Oh, my God. 
I have the main functionality, file browsing does not work. I click import, and nothing, nothing at all. The interface just ignores everything. No error, no success, nothing. 🥴
I also noticed that react does not run rerender. Most likely, some variable is not in the dependencies of some react hook.

**Comment by MattWong-ca** [comment]
Hey @SgtPooki, interested in working on this! To confirm in terms of UX, would the below flow work?

Click 'Import' --> Choose 'Bulk Import' (new option) --> user selects .txt or .csv file --> new imported files show up on the page

**Comment by asigalov61** [comment]
@SgtPooki @MattWong-ca Thank you very much for wanting to add this much needed feature! Looking forward to the final implementation :)

**Comment by agmap** [comment]
Many thanks in advance.
I really miss such a function.
Perhaps a bulk import of locally stored CAR files can also be integrated?

**Comment by MattWong-ca** [comment]
I still need to clean up the code a bit, but here's a functional prototype. 

Should I add an intermediate pop up that lets the user know about the required .txt format?

cc @lidel @SgtPooki 

https://github.com/user-attachments/assets/72033ad7-a18e-4a8c-9128-431f44e4cb27

**Comment by asigalov61** [comment]
@SgtPooki @MattWong-ca 

Thank you so much. guys! Can't wait for final implementation! :)

**Comment by asigalov61** [comment]
@SgtPooki @MattWong-ca Any updates on integration into IPFS release? Sorry to bother you so often but I just can't wait for this feature because available :)

Thank you.

**Comment by MattWong-ca** [comment]
Feature has been merged in #2307!

cc @SgtPooki good to close this issue

**Comment by asigalov61** [comment]
@MattWong-ca Thank you very much!

**Comment by SgtPooki** [comment]
Thanks a ton for your work here @MattWong-ca :)
